ethereum-contracts/README.md

```diff
@@ -1,72 +1,61 @@
-    - `$ testrpc`
-    - `$ truffle test` on a separate tab
+### Non-Goals of the Smart Contracts
+1. Creating a production-grade system for cross-chain value transfers which enforces strict permissions and limits access to locked funds.
+2. Implementing a validator set which enables observers to submit proof of fund locking transactions on Cosmos to Peggy. These features are not required for unidirectional transfers from Ethereum to Cosmos and will be re-integrated during phase two of the project, which aims to send funds from Cosmos back to Ethereum.
+3. Fully gas optimize and streamline operational functionality; ease and clarity of testing has been favored over some gas management and architectural best practices.
 
+## Ethereum Cosmos Bridge Architecture
+Unidirectional Peggy focuses on core features for unidirectional transfers. This prototype includes functionality to safely lock and unlock Ethereum and ERC20 tokens, emitting associated events which provide the Cosmos based Relayer and Oracle modules with the information required to complete unidirectional transfers. Validators witness the events and submit proof in the form of signed hashes to the Cosmos based modules, which are responsible for aggregating and tallying the Validators’ signatures and their respective signing power. The system is managed by the contract's deployer, designated internally as the relayer, a trusted third-party which can unlock funds and return them their original sender. If the contract’s balances under threat, the relayer can pause the system, temporarily preventing users from depositing additional funds.
 
-2) **Ganache**:
-    - `$ cd ethereum-contracts`
-    - Delete *build* folder: `$ rm -r build/`
-    - `$ truffle migrate --reset --compile-all`
-    - Open Ganache app, open the *Settings* button and make sure it's running on port **_7545_**.
-    - `$ truffle test --network ganache` on a separate tab
+The Peggy Smart Contract is deployed on the Ropsten testnet at address: 0x3de4ef81Ba6243A60B0a32d3BCeD4173b6EA02bb
 
+### Architecture Diagram
+![peggyarchitecturediagram](https://user-images.githubusercontent.com/15370712/58388886-632c7700-7fd9-11e9-962e-4e5e9d92c275.png)
 
-![alt text](./img/ganache_setup.png "Ganache Setup")
+### System Process:
+1. Users lock Ethereum or ERC20 tokens on the Peggy contract, resulting in the emission of an event containing the created item's original sender's Ethereum address, the intended recipient's Cosmos address, the type of token, the amount locked, and the item's unique nonce.
+2. Cosmos based Validators witness these lock events via a Relayer module and sign a hash containing the unique item's information, which is relayed from the Cosmos Relayer module to the Oracle module.
+3. Once the Oracle module has verified that the validators' aggregated signing power is greater than the specified threshold, it mints the appropriate amount of tokens and forwards them to the intended recipient.
```

Comment:
Is the idea here that the Relayer and Oracle module exist on the same chain? Would be helpful to explain that separation more clearly.

Reply:
Updated with: 'The Relayer is a service which interfaces with both blockchains, allowing validators to attest on the Cosmos blockchain that specific events on the Ethereum blockchain have occurred. The Relayer listens for `LogLock` events, parses information associated with the Ethereum transaction, uses it to build unsigned Cosmos transactions, and enables validators to sign and send the transactions to the Oracle module on Cosmos.'